cdap-ui/app/features/tracker/controllers/usage-ctrl.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class TrackerUsageController {
+  constructor($state, $scope, myTrackerApi) {
+
+    this.$state = $state;
+    this.$scope = $scope;
+    this.myTrackerApi = myTrackerApi;
+
+    let entitySplit = this.$state.params.entityType.split(':');
+
+    this.entityType = entitySplit;
+
+    this.timeRangeOptions = [
+      {
+        label: 'Last 7 days',
+        start: 'now-7d',
+        end: 'now'
+      },
+      {
+        label: 'Last 14 days',
+        start: 'now-14d',
+        end: 'now'
+      },
+      {
+        label: 'Last month',
+        start: 'now-30d',
+        end: 'now'
+      },
+      {
+        label: 'Last 6 months',
+        start: 'now-180d',
+        end: 'now'
+      },
+      {
+        label: 'Last 12 months',
+        start: 'now-365d',
+        end: 'now'
+      }
+    ];
+
+    this.timeRange = {
+      start: $state.params.start || 'now-7d',
+      end: $state.params.end || 'now'
+    };
+
+    this.customTimeRange = {
+      startTime: null,
+      endTime: null
+    };
+
+    this.selectedTimeRange = this.findTimeRange();
+
+    this.fetchAuditHistogram();
+    this.fetchTopDatasets();
+    this.fetchTimeSince();
+  }
+
+  findTimeRange() {
+    let match = this.timeRangeOptions.filter( (option) => {
+      return option.start === this.timeRange.start && option.end === this.timeRange.end;
+    });
+
+    if (match.length === 0) {
+      this.isCustom = true;
+      this.customTimeRange.startTime = new Date(parseInt(this.$state.params.start, 10) * 1000);
```

Comment:
We should probably have a check if `$state.params.start & end` are valid and non-null dates.